tests/test_machine.py

```diff
@@ -5,64 +5,50 @@
 
 import pytest
 
-import mkosi.machine as machine
 from mkosi.backend import MkosiException
+from mkosi.machine import Machine, MkosiMachineTest
 
 pytestmark = [
     pytest.mark.integration,
-    pytest.mark.parametrize("verb", ["boot", "qemu"]),
     pytest.mark.skipif(os.getuid() != 0, reason="Must be invoked as root.")
 ]
 
+class MkosiMachineTestCase(MkosiMachineTest):
+    def test_simple_run(self) -> None:
+        process = self.machine.run(["echo", "This is a test."])
+        assert process.stdout.strip("\n") == "This is a test."
 
-def test_simple_run(verb: str) -> None:
-    with machine.Machine([verb]) as m:
-        p = m.run(["echo", "This is a test."])
-        assert "This is a test." == p.stdout.strip("\n")
-
-    assert m.exit_code == 0
-
-
-def test_wrong_command(verb: str) -> None:
-    # First tests with argument check = True from mkosi.backend.run(), therefore we see if an exception is raised
-    with machine.Machine([verb]) as m:
+    def test_wrong_command(self) -> None:
+        # Check = True from mkosi.backend.run(), therefore we see if an exception is raised
         with pytest.raises(MkosiException):
-            m.run(["NonExisting", "Command"])
+            self.machine.run(["NonExisting", "Command"])
         with pytest.raises(MkosiException):
-            m.run(["ls", "NullDirectory"])
-
-    assert m.exit_code == 0
+            self.machine.run(["ls", "NullDirectory"])
 
-    # Second group of tests with check = False to see if stderr and returncode have the expected values
-    with machine.Machine([verb]) as m:
-        result = m.run(["NonExisting", "Command"], check=False)
+        # Check = False to see if stderr and returncode have the expected values
+        result = self.machine.run(["NonExisting", "Command"], check=False)
         assert result.returncode in (203, 127)
 
-        result = m.run(["ls", "-"], check=False)
+        result = self.machine.run(["ls", "-"], check=False)
         assert result.returncode == 2
         assert "No such file or directory" in result.stderr
 
-    assert m.exit_code == 0
-
-
-def test_infinite_command(verb: str) -> None:
-    with machine.Machine([verb]) as m:
+    def test_infinite_command(self) -> None:
         with pytest.raises(TimeoutExpired):
-            m.run(["tail", "-f", "/dev/null"], 2)
-
-    assert m.exit_code == 0
+            self.machine.run(["tail", "-f", "/dev/null"], 2)
 
 
-def test_before_boot(verb: str) -> None:
-    m = machine.Machine([verb])
+def test_before_boot() -> None:
+    m = Machine(["boot"])
```

Comment:
We can remove the explicit verb passing here and below now that we assign a default value in Machine's constructor